Sources/Apollo/GraphQLFirstReceivedAtTracker.swift

```diff
@@ -1,24 +1,24 @@
 import Foundation
 
 final class GraphQLFirstReceivedAtTracker: GraphQLResultAccumulator {
-  func accept(scalar: JSONValue, firstReceivedAt: Date, info: GraphQLResolveInfo) throws -> Date {
+  func accept(scalar: JSONValue, firstReceivedAt: Date, info: GraphQLResolveInfo) throws -> Date? {
     return firstReceivedAt
   }
 
-  func acceptNullValue(firstReceivedAt: Date, info: GraphQLResolveInfo) throws -> Date {
+  func acceptNullValue(firstReceivedAt: Date, info: GraphQLResolveInfo) throws -> Date? {
     return firstReceivedAt
   }
 
-  func accept(list: [Date], info: GraphQLResolveInfo) throws -> Date {
-    return list.min() ?? .distantPast
+  func accept(list: [Date?], info: GraphQLResolveInfo) throws -> Date? {
+    return list.compactMap({ $0 }).max()
```

Comment:
maybe add a comment somewhere around on why we are using `max`?

Reply:
Done, thanks for the reminder